addon-test-support/scroll.js

```diff
@@ -0,0 +1,47 @@
+import getElementWithAssert from './-private/get-element-with-assert';
+import wait from 'ember-test-helpers/wait';
+
+
+/*
+  Triggers a paint (and therefore flushes any queued up scroll events).
+
+  @method triggerFlushWithPromise
+  @return {RSVP.Promise}
+  @private
+*/
+function triggerFlushWithPromise(){
+  return wait().then(() => {
```

Comment:
@pzuraq I'd lie if I told your that I got it. So in short, is this kind of approach needed for people adding `scroll` events that trigger async behaviour?
If so, which one is more correct?
```js
return new RSVP.Promise(function(resolve) { window.requestAnimationFrame(resolve) }).then(() => wait())
```
or
```js
return wait().then(() => new RSVP.Promise(function(resolve) { window.requestAnimationFrame(resolve) }));
```